mlocal/frags/Makefile.stub

```diff
@@ -194,7 +194,9 @@ collect:
 check:
 	@echo " CHECK go fmt"
 	$(V)export CGO_CPPFLAGS="$(cgo_CPPFLAGS)" CGO_LDFLAGS="$(cgo_LDFLAGS)" && \
-		cd $(SOURCEDIR) && test -z $(go fmt ./...)
+		cd $(SOURCEDIR) && \
+		( test -z "`go fmt -n ./... | sed 's/ -w / /' | sh | tee /dev/stderr`" || \
```

Comment:
```
test -z "`gofmt -l \`go list -f "{{.Dir}}" ./...\``"
```

Reply:
I had tried `go list -f {{.Dir}}` but the problem is that it lists subdirectories of other directories so some files get checked twice.  I posted an alternative in a [comment on #2011](https://github.com/sylabs/singularity/pull/2011#issuecomment-422538900) that filters the output with awk but it isn't pretty.  I think the form I've got now is simpler.

Reply:
I was able to reproduce this behavior, but I would really prefer to not use linux text processing to handle this so we can minimize the chance we make human error. I've come up with a workaround which sticks within the `go` to minimize error:

```
mibauer@ubuntu:~/go/src/github.com/sylabs/singularity$ gofmt -l `go list -f '{{$d := .Dir}} {{ range .GoFiles }}{{$d}}/{{.}} {{end}}' ./...`
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/build.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/capability.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/capability_add.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/capability_drop.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/capability_list.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/instance.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/instance_start.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/keys.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/keys_list.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/keys_newpair.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/keys_pull.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/keys_push.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/keys_search.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/push.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/sign.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/cmd/singularity/cli/verify.go
/home/mibauer/go/src/github.com/sylabs/singularity/src/pkg/syecl/syecl.go
```

`go list [-f tmpl]` is a fully functional [template](https://golang.org/pkg/text/template/) meaning I was able to get `go list` to output a list of every single file's absolute path and `gofmt -l` can take those paths in. That seems to solve the recursion issue you caught and not forcing us to use `awk`!

Reply:
That's impressive, but are you sure you like that better than `go fmt -n ./... | sed 's/ -w / /' | sh`?  I think the go fmt version is a quite a bit easier to understand.  It's certainly a lot fewer characters.  If you're sure, I will make the change. 

Oh, the file list might get too long for the exec buffer on some OS versions.  Probably it would be better to pipe the output of go list into `xargs gofmt -l`.

Reply:
I see that the go list form also lists complete path names instead of relative, which I think is a disadvantage for readability.